stdlib/public/runtime/SwiftValue.mm

```diff
@@ -63,65 +64,151 @@ - (id)copyWithZone:(NSZone *)zone;
   /// The type of the value contained in the `__SwiftValue` box.
   const Metadata *type;
 
-  /// The base type that introduces the `Hashable` conformance.
-  /// This member is only available for native Swift errors.
+  /// The base type that introduces the `Hashable` or `Equatable` conformance.
   /// This member is lazily-initialized.
-  /// Instead of using it directly, call `getHashableBaseType()`.
-  mutable std::atomic<const Metadata *> hashableBaseType;
+  /// Instead of using it directly, call `getHashableBaseType()` or `getEquatableBaseType()`
+  mutable std::atomic<const Metadata *> cachedBaseType;
```

Comment:
Making this a `const void *` or `uintptr_t` would help enforce the requirement described in the comment. I'd say make both this and `cachedConformance` `uintptr_t` since that's the final type you end up dealing with when getting or setting a value.

Reply:
Ah...  Yes, that's a good point.  Since this isn't _really_ a pointer (it's an integer value that encodes a pointer), `uintptr_t` is the more correct type, and it should reduce some of the casting shenanigans below.

Reply:
Done.